registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,563 @@
+		Prefix: d.swiftPath(path) + "/",
+	}
+
+	objects, err := d.Conn.ObjectsAll(d.Container, &opts)
+	if err != nil {
+		return parseError(path, err)
+	}
+
+	if d.BulkDeleteSupport {
+		filenames := make([]string, len(objects))
+		for i, obj := range objects {
+			filenames[i] = obj.Name
+		}
+		if _, err := d.Conn.BulkDelete(d.Container, filenames); err != swift.Forbidden {
+			return parseError(path, err)
+		}
+	}
+
+	for _, obj := range objects {
+		if obj.PseudoDirectory {
+			continue
+		}
+		if _, headers, err := d.Conn.Object(d.Container, obj.Name); err == nil {
+			manifest, ok := headers["X-Object-Manifest"]
+			if ok {
+				components := strings.SplitN(manifest, "/", 2)
+				segContainer := components[0]
+				segments, err := d.getAllSegments(components[1])
+				if err != nil {
+					return parseError(obj.Name, err)
+				}
+
+				for _, s := range segments {
+					if err := d.Conn.ObjectDelete(segContainer, s.Name); err != nil {
+						return parseError(s.Name, err)
+					}
+				}
+			}
+		} else {
+			return parseError(obj.Name, err)
+		}
+
+		if err := d.Conn.ObjectDelete(d.Container, obj.Name); err != nil {
+			return parseError(obj.Name, err)
+		}
+	}
+
+	if _, err := d.Stat(ctx, path); err == nil {
+		return parseError(path, d.Conn.ObjectDelete(d.Container, d.swiftPath(path)))
+	} else if len(objects) == 0 {
+		return storagedriver.PathNotFoundError{Path: path}
+	}
+
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
+	return "", storagedriver.ErrUnsupportedMethod
+}
+
+func (d *driver) swiftPath(path string) string {
+	return strings.TrimLeft(strings.TrimRight(d.Prefix+"/files"+path, "/"), "/")
+}
+
+func (d *driver) swiftSegmentPath(path string) string {
+	return strings.TrimLeft(strings.TrimRight(d.Prefix+"/segments"+path, "/"), "/")
+}
+
+func (d *driver) getContentType() string {
+	return "application/octet-stream"
+}
+
+func (d *driver) getAllSegments(path string) ([]swift.Object, error) {
+	return d.Conn.ObjectsAll(d.Container, &swift.ObjectsOpts{Prefix: d.swiftSegmentPath(path)})
+}
+
+func (d *driver) createManifest(path string) (*swift.ObjectCreateFile, error) {
+	headers := make(swift.Headers)
+	headers["X-Object-Manifest"] = d.Container + "/" + d.swiftSegmentPath(path)
+	return d.Conn.ObjectCreate(d.Container, d.swiftPath(path), false, "",
+		d.getContentType(), headers)
+}
+
+func detectBulkDelete(authURL string) (bulkDelete bool) {
+	resp, err := http.Get(gopath.Join(authURL, "..", "..") + "/info")
+	if err == nil {
+		defer resp.Body.Close()
+		decoder := json.NewDecoder(resp.Body)
+		var infos swiftInfo
+		if decoder.Decode(&infos) == nil {
+			_, bulkDelete = infos["bulk_delete"]
+		}
+	}
+	return
+}
+
+func parseError(path string, err error) error {
```

Comment:
@stevvooe I pushed https://github.com/nevermosby/distribution/commit/1ff25386d2896484971523c21c9f73a7dcda2a9c that inline the error handling